### ISSUE TITLE
OpenCPU request failure

### ISSUE THREAD

**Issue by flammit** [jeroenooms/opencpu#142]
I'm intermittently getting into a state where all calls to OpenCPU result in the following error:

```
cannot popen '/usr/bin/which 'uname' 2>/dev/null', probable reason 'Resource temporarily unavailable'

In call:
system(paste(which, shQuote(names[i])), intern = TRUE, ignore.stderr = TRUE)
```

Any ideas as to why this could be happening?  Restarting the system seems to fix this, but I'd like to avoid doing that if possible.

Thanks!

**Comment by jeroenooms** [comment]
Is the machine running out of disk space perhaps? What hw are you running
on?

**Comment by flammit** [comment]
We're seeing the problem on a clean Ubuntu 14.04.1LTS on AWS with opencpu just installed.  It has plenty of space on all volumes (including /tmp).

**Comment by jeroenooms** [comment]
Very strange. Anything in /var/log/kern.log ?

**Comment by flammit** [comment]
Agreed - very strange.  Nothing in /var/log/kern.log.  It dies about 80 requests or so in... Not sure if there's a process / fork limit that's not being cleaned up.

**Comment by jeroenooms** [comment]
Are you using any crazy R packages? Also is your ubuntu system up to date?

```bash
sudo apt-get update
sudo apt-get dist-upgrade
```

**Comment by flammit** [comment]
Yeah the system was updated just now to latest on spin up before the opencpu install.  No weird R packages (at least not weirder than an image that is up and running fine).

As a workaround, we're going to try to go back to an older OS image and just update our stats packages instead.

**Comment by jeroenooms** [comment]
The OS should not be the problem though, I'm running Ubuntu Trusty everywhere (and keeping it up to date).

**Comment by jeroenooms** [comment]
Which Ubuntu EC2 image did you use? I always use these: http://cloud-images.ubuntu.com/trusty/current/

**Comment by jeroenooms** [comment]
Did you end up resolving this issue?

**Comment by flammit** [comment]
Thanks for checking in.  I have not, but I also haven't had the time to dig too much into this.  I'm going to close this issue out and try this again using the latest trusty image as suggested and revert with a new issue if I'm still having problems (we were using a 14.04.1 image with update && upgrade).

**Comment by moosilauke18** [comment]
I am getting this error now.  I suspect it has something to do with a lack of either CPU or memory, but I am unsure. I am running it in the opencpu/base docker container on kubernetes.

**Comment by jeroenooms** [comment]
Are there any messages in `/var/log/apache2/error.log` or `/var/log/opencpu/error.log` or `/var/log/kern.log` ?

**Comment by moosilauke18** [comment]
`/var/log/apache2/error.log` has things in it.  `kern.log` doesn't exist.  
In `/var/log/apache2/error.log`

```
Setting locale to en_US.UTF-8
Checking Apparmor Status...
 AppArmor status lookup failed. Error code: ENOSYS
AppArmor extensions to the system are not available.
AppArmor not available. Running OpenCPU without security profile but with rlimits.
Setting locale to en_US.UTF-8
Setting locale to en_US.UTF-8
Checking Apparmor Status...
 AppArmor status lookup failed. Error code: ENOSYS
AppArmor extensions to the system are not available.
AppArmor not available. Running OpenCPU without security profile but with rlimits.
Checking Apparmor Status...
 AppArmor status lookup failed. Error code: ENOSYS
AppArmor extensions to the system are not available.
AppArmor not available. Running OpenCPU without security profile but with rlimits.
OpenCPU cloud server ready.
OpenCPU cloud server ready.
OpenCPU cloud server ready.
```

**Comment by moosilauke18** [comment]
It's also important to note, that it only occurs when running on the kubernetes cluster.  If I run the docker image on my local machine, I have no problems.

**Comment by moosilauke18** [comment]
Ok, so I've played a little more with it.  It seems to be a problem with too little memory.  I gave the container more memory and it seems to alleviate the problem.  

If I use `1Gi` or less of memory for the container it throws the above error. It also sometimes gives the error `R call did not return within 90 seconds. Terminating process.`  If I give it around 1.25Gi or more of memory it solves the issue. 

If I give it even less memory like 500Mi it throws even different errors such as `R call failed: process died`.

 My guess is the OP of the issue was using something like a `t2.micro` aws instance that heavily limits the memory causing it to generate the error.
